topic-operator/src/test/java/io/strimzi/operator/topic/v2/LogCaptor.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.v2;
+
+import io.strimzi.operator.common.ReconciliationLogger;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.filter.LevelRangeFilter;
+import org.apache.logging.log4j.spi.AbstractLogger;
+import org.apache.logging.log4j.spi.ExtendedLogger;
+import org.apache.logging.log4j.spi.ExtendedLoggerWrapper;
+
+import java.lang.reflect.Field;
+import java.util.Locale;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * A mechanism for asserting that messages are logged, or not logged, through Log4j2.
+ * <pre>
+ * {@code
+ *         try (var captor = LogCaptor.requireMatch(LOGGER, Level.INFO,
+ *                 le -> "Hello, world".equals(le.getMessage().getFormattedMessage()),
+ *                 50, TimeUnit.MILLISECONDS)) {
+ *             LOGGER.info("Hello, world");
+ *         }
+ * }
+ * </pre>
+ */
+public class LogCaptor implements AutoCloseable {
```

Comment:
Am I wrong or this class is never used in the UTO logic? I see it just used in the corresponding tests.

Reply:
You're correct, and that's why it's in `topic-operator/src/test/java`.